File: 2025/F/answers/java/Solution.java
import java.util.*;
import java.io.*;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNext()) return;
        String s = sc.next();
        int n = s.length();
        
        Map<Long, Long> factors = new TreeMap<>();
        if (n <= 15) {
            long val = Long.parseLong(s);
            factorize(val, factors);
        } else {
            // Take first 14 digits to keep sqrt(P) small enough for trial division
            long p = Long.parseLong(s.substring(0, 14));
            long l = n - 14;
            factorize(p, factors);
            factors.put(2L, factors.getOrDefault(2L, 0L) + l);
            factors.put(5L, factors.getOrDefault(5L, 0L) + l);
        }
        
        List<Map.Entry<Long, Long>> res = new ArrayList<>();
        for (Map.Entry<Long, Long> entry : factors.entrySet()) {
            if (entry.getValue() > 0) res.add(entry);
        }
        
        System.out.println(res.size());
        for (Map.Entry<Long, Long> entry : res) {
            System.out.println(entry.getKey() + " " + entry.getValue());
        }
    }

    static void factorize(long n, Map<Long, Long> factors) {
        for (long i = 2; i * i <= n; i++) {
            while (n % i == 0) {
                factors.put(i, factors.getOrDefault(i, 0L) + 1);
                n /= i;
            }
        }
        if (n > 1) factors.put(n, factors.getOrDefault(n, 0L) + 1);
    }
}

File: 2025/F/answers/kt/Solution.kt
import java.util.*

fun factorize(n: Long, factors: MutableMap<Long, Long>) {
    var temp = n
    var i = 2L
    while (i * i <= temp) {
        while (temp % i == 0L) {
            factors[i] = (factors[i] ?: 0L) + 1L
            temp /= i
        }
        i++
    }
    if (temp > 1) factors[temp] = (factors[temp] ?: 0L) + 1L
}

fun main() {
    val sc = Scanner(System.`in`)
    if (!sc.hasNext()) return
    val s = sc.next()
    val n = s.length

    val factors = TreeMap<Long, Long>()
    if (n <= 15) {
        val value = s.toLong()
        factorize(value, factors)
    } else {
        // Take first 14 digits for efficient trial division
        val p = s.substring(0, 14).toLong()
        val l = (n - 14).toLong()
        factorize(p, factors)
        factors[2L] = (factors[2L] ?: 0L) + l
        factors[5L] = (factors[5L] ?: 0L) + l
    }

    val res = factors.filter { it.value > 0 }
    println(res.size)
    for ((p, e) in res) {
        println("$p $e")
    }
}
